apps/console/src/features/claims/pages/claim-dialects.tsx

```diff
@@ -200,316 +231,376 @@ const ClaimDialectsPage: FunctionComponent<ClaimDialectsPageInterface> = (
-                                                    width={ 4 }
-                                                    verticalAlign="middle"
-                                                    textAlign="right"
-                                                    data-testid={ `${ testId }-local-dialect-` +
-                                                            "attributes-edit-icon" }
-                                                >
-                                                    <Popup
-                                                        content={
-                                                            hasRequiredScopes(
-                                                                featureConfig?.attributeDialects,
-                                                                featureConfig?.attributeDialects?.
-                                                                    scopes?.create,
-                                                                allowedScopes
-                                                            ) ?
-                                                                t("common:edit") :
-                                                                t("common:view")
-                                                        }
-                                                        trigger={
-                                                            hasRequiredScopes(
-                                                                featureConfig?.attributeDialects,
-                                                                featureConfig?.attributeDialects?.
-                                                                    scopes?.create,
-                                                                allowedScopes
-                                                            ) ?
-                                                                <Icon color="grey" name="pencil" /> :
-                                                                <Icon color="grey" name="eye" />
-                                                        }
-                                                        inverted
-                                                    />
-                                                </Grid.Column>
-                                            </Grid.Row>
-                                        </Grid>
-                                    </List.Item>
-                                </List>
-                            </EmphasizedSegment>
-                        </>
-                    )
+                    { 
+                        hasRequiredScopes(
+                            featureConfig?.attributeDialects,
+                            featureConfig?.attributeDialects?.scopes?.read,
+                            allowedScopes
+                        ) && (
+                            isLoading 
```

Comment:
Shouldn't `isLoading` be outside the `hasRequiredScopes` check? In the current way, nothing will be rendered until the feature config is resolved, but if we move `isLoading` outside, the loading placeholders will be shown until the feature config is resolved.

Please correct me if I'm missing something obvious though :)

Reply:
1. `hasRequiredScopes` check is not a async check. IMO there is no point of rendering a loader for such cases.
2. If we move isLoading outside, there will be a loading placeholder (until the API call is resolved) even though there is no required scopes.

Reply:
> 1. `hasRequiredScopes` check is not a async check. IMO there is no point of rendering a loader for such cases.

Then, +1 not to move `isLoading` outside.

> 2\. there will be a loading placeholder (until the API call is resolved) even though there is no required scopes.

In the current code, are API calls made from the FE without validating scopes? IMO, we can do some improvements there. However, considering the scope of this PR, we can consider this in a separate effort.